src/cli/main.rs

```diff
@@ -341,12 +356,28 @@ fn format(opt: opt::Opt) -> Result<i32> {
                     let tx = tx.clone();
                     let opt = opt.clone();
 
+                    let should_skip_format = match &opt.stdin_filepath {
+                        Some(filepath) => {
+                            let ignore = get_ignore(filepath.parent().unwrap()).unwrap();
+                            match ignore.matched(&filepath, false) {
+                                ignore::Match::Ignore(_) => true,
+                                _ => false,
+                            }
+                        }
+                        None => false,
+                    };
+
```

Comment:
Instead of computing this and still calling `format_string`, could we potentially just exit early here?
```suggestion
                    match &opt.stdin_filepath {
                        Some(filepath) => {
                            let ignore = get_ignore(filepath.parent().unwrap()).unwrap();
                            match ignore.matched(&filepath, false) {
                                ignore::Match::Ignore(_) => continue,
                                _ => (),
                            }
                        }
                        None => (),
                    };
```

And then the changes to `format_string` are no longer necessary as well

Reply:
If its changed to this, you could probably also rewrite it as an `if let` statement

Reply:
> Instead of computing this and still calling format_string, could we potentially just exit early here?

Format string is doing some other things, e.g. diffing, wrapping with `FormatResult::*`, converting to bytes etc.

Is it okay to bypass all that?

Reply:
Ah, good point. Not sure what is better to do here, so it's fine to leave it for now, thanks